### DISCUSSION TITLE
Migrating custom QInput Component from v1 to v2

### DISCUSSION THREAD

**Discussion by Dirk-** [quasarframework/quasar#11944] [General - Components / Directives / etc] [answered]
In my v1 app I have some preconfigured, reusable components based on QInput, QSelect etc. which I use on several pages and dialogs. I learned from Danny Connell's Udemy course, so the QInput based v1 component looks similar to this:

```
<template>
  <div class="row q-mb-sm">
    <q-input
      :value="name"
      :rules="[val => !!val || 'Required']"
      @input="$emit('update:name', $event)"
      @focus="selectAll($event)"
      class="col"
      outlined
      clearable
      ref="name"
      label="Name" />
  </div>
</template>

<script>
export default {
  props: ['name'],
  methods: {
    selectAll (event) {
      event.target.select()
    }
  }
}
</script>
```

I am using it in several QForms like this:

```
<modal-task-name
  :name.sync="taskToSubmit.name"
  ref="modalTaskName" />
```

I am struggling with the [v-model migration](https://quasar.dev/start/upgrade-guide#quasar-components). I followed the migration guide to the point where my app compiles, has no linting errors and starts, but I cannot figure out how to get values from my custom components to the parent component.

I replaced `:value` with `:model-value`, `@input` with `@update:model-value` and `:name.sync` with `v-model`, it does not work.  I looked at the Vue guide [Using v-model on components](https://v3.vuejs.org/guide/component-basics.html#using-v-model-on-components) and tried it like this:

```
<template>
  <div class="row q-mb-sm">
    <q-input
      :value="modelValue"
      :rules="[val => !!val || 'Required']"
      @input="$emit('update:modelValue', $event.target.value)"
      @focus="selectAll($event)"
      class="col"
      outlined
      clearable
      ref="name"
      label="Name" />
  </div>
</template>

<script>
export default {
  props: ['modelValue'],
  emits: ['update:modelValue'],
  methods: {
    selectAll (event) {
      event.target.select()
    }
  }
}
</script>
```
using it like this:
```
<modal-task-name
  v-model="taskToSubmit.name"
  ref="modalTaskName" />
```

I am sure I did not fully understand the principle. How can I migrate the component so that I can use it with `v-model`?

**Comment by metalsadman** [comment, marked as answer]
`:value` -> `:model-value`, `@input` -> `@update:model-value`. if you're not going to do anything special (passing extra values etc...), then `v-bind="$attrs"` is enough (this is needed if your wrapper is not the root element along with `inheritAttrs: false`).
example using simple wrapper and your implementation.
https://codepen.io/metalsadman/pen/oNGMRQV?editors=1010
